presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduClientSession.java

```diff
@@ -138,43 +137,62 @@ public Schema getTableSchema(KuduTableHandle tableHandle)
         return KuduTableProperties.toMap(table);
     }
 
-    public List<KuduSplit> buildKuduSplits(KuduTableLayoutHandle layoutHandle)
+    public List<KuduSplit> buildKuduSplits(KuduTableHandle tableHandle)
     {
-        KuduTableHandle tableHandle = layoutHandle.getTableHandle();
         KuduTable table = tableHandle.getTable(this);
         final int primaryKeyColumnCount = table.getSchema().getPrimaryKeyColumnCount();
         KuduScanToken.KuduScanTokenBuilder builder = client.newScanTokenBuilder(table);
 
-        TupleDomain<ColumnHandle> constraintSummary = layoutHandle.getConstraintSummary();
-        if (!addConstraintPredicates(table, builder, constraintSummary)) {
+        TupleDomain<ColumnHandle> constraint = tableHandle.getConstraint();
+        if (!addConstraintPredicates(table, builder, constraint)) {
             return ImmutableList.of();
         }
 
-        Optional<Set<ColumnHandle>> desiredColumns = layoutHandle.getDesiredColumns();
-        if (desiredColumns.isPresent()) {
-            if (desiredColumns.get().contains(KuduColumnHandle.ROW_ID_HANDLE)) {
```

Comment:
Where did this logic go? If I understood it correctly, if one of the requested columns was the magic ROW ID column, it would include the underlying columns corresponding to the primary key. I don't see that anymore except for the delete case. Was that column only used in that scenario?

Reply:
Yes, the `ROW_ID_HANDLE` column handle is only used in delete.

I changed the condition here to `tableHandle.isDeleteHandle()` because unlike the original `KuduTableLayoutHandle`, `KuduTableHandle.desiredColumns` won't include `ROW_ID_HANDLE`.